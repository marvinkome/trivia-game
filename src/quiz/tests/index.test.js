import React from 'react';
import { shallow } from 'enzyme';
import { Quiz } from '../index';
import Body from '../body';

describe('quiz page tests', () => {
    it('renders correctly', () => {
        const props = {
            quiz: {
                results: []
            }
        };
        const wrapper = shallow(<Quiz {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders body correctly', () => {
        const wrapper = shallow(<Body quiz={[]} />);
        expect(wrapper).toMatchSnapshot();
    });
});
