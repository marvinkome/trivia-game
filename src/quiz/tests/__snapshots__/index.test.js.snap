// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quiz page tests renders body correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Body
    quiz={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <QuizCard />,
        <p
          className="answer"
        >
          <a
            className="btn"
            onClick={[Function]}
          >
            True
          </a>
          <a
            className="btn"
            onClick={[Function]}
          >
            False
          </a>
        </p>,
        <p
          className="quiz-counter"
        >
          1
          /
          0
        </p>,
        <a
          className="btn"
          id="go-back"
          onClick={[Function]}
        >
          Previous Question
        </a>,
        <a
          className="btn"
          id="go-front"
          onClick={[Function]}
        >
          Skip question
        </a>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "item": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="btn"
              onClick={[Function]}
            >
              True
            </a>,
            <a
              className="btn"
              onClick={[Function]}
            >
              False
            </a>,
          ],
          "className": "answer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "True",
              "className": "btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "True",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "False",
              "className": "btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "False",
            "type": "a",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            1,
            "/",
            0,
          ],
          "className": "quiz-counter",
        },
        "ref": null,
        "rendered": Array [
          1,
          "/",
          0,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Previous Question",
          "className": "btn",
          "id": "go-back",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Previous Question",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Skip question",
          "className": "btn",
          "id": "go-front",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Skip question",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <QuizCard />,
          <p
            className="answer"
          >
            <a
              className="btn"
              onClick={[Function]}
            >
              True
            </a>
            <a
              className="btn"
              onClick={[Function]}
            >
              False
            </a>
          </p>,
          <p
            className="quiz-counter"
          >
            1
            /
            0
          </p>,
          <a
            className="btn"
            id="go-back"
            onClick={[Function]}
          >
            Previous Question
          </a>,
          <a
            className="btn"
            id="go-front"
            onClick={[Function]}
          >
            Skip question
          </a>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "item": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="btn"
                onClick={[Function]}
              >
                True
              </a>,
              <a
                className="btn"
                onClick={[Function]}
              >
                False
              </a>,
            ],
            "className": "answer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "True",
                "className": "btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "True",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "False",
                "className": "btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "False",
              "type": "a",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              1,
              "/",
              0,
            ],
            "className": "quiz-counter",
          },
          "ref": null,
          "rendered": Array [
            1,
            "/",
            0,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Previous Question",
            "className": "btn",
            "id": "go-back",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Previous Question",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Skip question",
            "className": "btn",
            "id": "go-front",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Skip question",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`quiz page tests renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Quiz
    quiz={
      Object {
        "results": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "quiz": Array [],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "quiz": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
